src/utlist.h

```diff
@@ -349,6 +349,26 @@ do {
   }                                                                                            \
 } while (0)
 
+#define LL_INSERT_INORDER(head,add,cmp)                                                        \
+  do {                                                                                         \
+    LDECLTYPE(head) _pos;                                                                      \
+    _CASTASGN(_pos, head);                                                                     \
+    LL_FIND_POS(_pos,add,cmp);                                                                 \
+    LL_APPEND_ELEM(head,_pos,add);                                                             \
+  } while (0)
+
+#define LL_FIND_POS(head,node,cmp)                                                             \
+  LL_FIND_POS2(head,node,cmp,next)                                                             \
+
+#define LL_FIND_POS2(head,node,cmp,next)                                                       \
+  do {                                                                                         \
+    for (; (head)->next != NULL; (head) = (head)->next ) {                                     \
+      if (cmp((head)->next, (node)) > 0) {                                                     \
```

Comment:
Nit: You don't need the parens around `node` here because it's being passed as an argument (so the only bad interaction it could have is if it had embedded commas, and macro arguments can't have embedded commas). Since the parens are unneeded, please remove them.

(Also note that if you wanted to avoid weird interactions with `cmp` you'd have to write `(cmp)`; but we actually *want* the weird interactions there, e.g. if `cmp` is the name of a two-argument macro then we'd like to expand the macro. So that's correct.)